### DISCUSSION TITLE
The selection of JSON library

### DISCUSSION THREAD

**Discussion by binbjz** [fastapi/fastapi#4547] [Questions] [answered]
I noticed in the source code that we are using OrJSON now, but I am a little worried, because orJSON takes a long time to feedback and repair problems, and the support is slow and not as good as ultraJSON support, so I want to know. Will we always use OrJSON? I think UJSON would be a better choice. It's just a concern.

**Comment by odiseo0** [comment, marked as answer]
orjson & ujson are optional dependencies

**Reply by tiangolo** [reply]
Yep. :point_up: 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Reply by simon-liebehenschel** [reply]
> orjson & ujson are optional dependencies

@tiangolo @odiseo0  In 2022 a situation with `orjson` could be a little better, but in 2024 I think it may be not the best idea to promote abandonware/scam, even  if it is "optional".

`orjson` maintainer ignores bugs. All issues are auto-closed by a Stale Bot:   
https://github.com/ijl/orjson/issues?q=is%3Aissue+is%3Aclosed

The same story is about Pull Requests:   
https://github.com/ijl/orjson/pulls?q=is%3Apr+is%3Aclosed

I understand that an open source maintainer has no obligations to us, but a respectful maintainer behavior is to say "_I cannot maintain this. Feel free to fork or use another library_".   

Due to the current behavior, I consider `orjson` as abandonware/scam because it fools other developers to think that this library is safe to use.

Would you like to open an issue to remove `orjson` and, optionally, consider a maintained library instead?
